mollie/orders.go

```diff
@@ -404,10 +407,13 @@ func (ors *OrdersService) CreateOrderPayment(orderID string, ordPay *OrderPaymen
 
 	res, err := ors.client.Do(req)
 	if err != nil {
-		err = json.Unmarshal(res.content, &errorResponse)
-		if err != nil {
-			return
+		if len(res.content) != 0 {
```

Comment:
I think in the `Response` struct you also have access to the full response, can we check if the Content-Lenght header is set? that way you don't need to stress the code by recalculating the byte length here.

It will also be good to move it to a separate function, that way we don't need to write the same code over and over.

Reply:
Yes, I can access the content length of the Response struct.
You mean like this?

```suggestion
res, err := ors.client.Do(req)
if err != nil {
	errorResponse, err = parseErrorResponse(res)
	if err != nil {
		return
	}
	return
}
```

```suggestion
func parseErrorResponse(res *Response) (errorResponse *ErrorResponse, err error) {
	if res.ContentLength > 0 {
		err = json.Unmarshal(res.content, &errorResponse)
		if err != nil {
			return
		}
	}

	return
}
```

And I think if I go with the same approach as the other services, when the endpoint return with Unprocessable Entity message (as the docs explained) the error response won't be handled because the response returns an object too, correct me if I'm wrong.

Reply:
I will say just go with the same approach than the other services, I will try to investigate and draft an issue for covering edge cases in the client from a general approach.

Probably during the weekend, I will try to cover some things reported in the other mollie clients that I feel might also become a problem here. 😄 

But overall the _test file looks good, I will say do not change the original implementation 🤔

Reply:
Okay, I will revert the changes in orders.go to the previous state, but it will fail on the HTTPRequestErrors test, is that okay?
Also I will include small changes in common_types.go regarding ErrorResponse struct, the Status type should be int not string, is that okay too?

Reply:
Status is the message we display, also in the response the Status is the message and StatusCode is the int value.

Reply:
No, not the Status prop in the Response. I'm talking about this
https://github.com/VictorAvelar/mollie-api-go/blob/ad83f76313692f8e03bf110a8312d49ef895ab56/mollie/common_types.go#L148

As explained in the [docs](https://docs.mollie.com/reference/v2/orders-api/create-order-payment) when the Response returned with Unprocessable Entity, it returns an object too. The Status in the Response is an int, whilst in the code it's a string. I will include the fix in the next push if it's alright?

Reply:
Yes, I get it but if you change that then all the other test cases are going to fail, ignore this case for now as it involves a common_type that might have an impact on the whole library.

If you can create an issue we can investigate for a solution together.

Reply:
Okay, then I will just revert the changes in orders.go